pkg/apis/core/v1beta1/defaults_shoot.go

```diff
@@ -193,6 +193,13 @@ func SetDefaults_Shoot(obj *Shoot) {
 	}
 }
 
+// SetDefaults_MaintenanceRotationConfig sets default values for MaintenanceRotationConfig objects.
+func SetDefaults_MaintenanceRotationConfig(obj *MaintenanceRotationConfig) {
```

Comment:
What about the `enabled` field? Don't we need to default it as well?

Reply:
IMO, there is no need to explicitly default to `false`, `nil` would work as `false` here as well.

Not defaulting the `enabled` field would also let us be more flexible if we decide in the future to make the field default to `true` for specific rotations.